src/Pos/Communication/Limits.hs

```diff
@@ -141,26 +158,48 @@ instance MessageLimited InnerSharesMap where
 instance MessageLimitedPure VssCertificate where
     msgLenLimit = 171
 
-instance MessageLimited (DataMsg GtMsgContents) where
-    type LimitType (DataMsg GtMsgContents) = ( Limit (DataMsg GtMsgContents)
-                                             , Limit (DataMsg GtMsgContents)
-                                             , Limit (DataMsg GtMsgContents)
-                                             , Limit (DataMsg GtMsgContents))
-    getMsgLenLimit _ = do
-        commLimit <-
-            fmap MCCommitment <$> getMsgLenLimit (Proxy @SignedCommitment)
-        sharesLimit <-
-            (MCShares <$> msgLenLimit <+>) <$>
+instance MessageLimited (DataMsg MCCommitment) where
+    type LimitType (DataMsg MCCommitment) =
+        Limit (DataMsg MCCommitment)
+    getMsgLenLimit _ =
+        fmap (DataMsg . MCCommitment)
+          <$> getMsgLenLimit (Proxy @SignedCommitment)
+
+instance MessageLimited (DataMsg MCOpening) where
+    type LimitType (DataMsg MCOpening) =
```

Comment:
It's not necessary, this is default. Also you could define instance `MessageLimitedPure`, write a pure test for it and then you would be able to write `instance MessageLimited (DataMsg MCOpening)` (without anything else).